File: src/main/java/gfs/FileMetadata.java
package gfs;

import java.util.ArrayList;

public class FileMetadata {
    String key;
    long size;
    boolean isDir;
    ArrayList<Chunk> chunks;

    public FileMetadata(String key) {
        this.key = key;
    }

    public FileMetadata(int size) {
        var numChunks = (int) Math.ceil(size / (double) FileSystem.CHUNK_SIZE);
        chunks = new ArrayList<>(numChunks);
    }

}

File: src/main/java/gfs/FileMetadata.kt
package gfs

import kotlin.math.ceil

class FileMetadata(val path: String, val isDir: Boolean, val size: Long)  {


    constructor() : this("", true, -1) {
    }

    companion object {
        fun newFile(path: String, size: Long): FileMetadata {
            return FileMetadata(path,false,size)
        }
    }
}
